client/app/lib/components/UnsubmitButton.jsx

```diff
@@ -42,8 +37,7 @@ UnsubmitButton.propTypes = {
   buttonId: PropTypes.number.isRequired,
   color: PropTypes.string.isRequired,
   isUnsubmitting: PropTypes.bool.isRequired,
-  setState: PropTypes.func.isRequired,
-  state: PropTypes.object.isRequired,
+  onClick: PropTypes.func.isRequired,
 };
 
 export default connect((state, ownProps) => ({
```

Comment:
Instead of getting the state directly from redux, can we pass the state down from the parent component instead?